postprocessing/comstockpostproc/comstock_query_builder.py

```diff
@@ -0,0 +1,328 @@
+                "building_type",
+                sum("total_site_gas_kbtu") AS "total_site_gas_kbtu",
+                sum("total_site_electricity_kwh") AS "total_site_electricity_kwh"
+            FROM
+            (
+                SELECT
+                    EXTRACT(MONTH from "time") as "month",
+                    SUBSTRING("build_existing_model.county_id", 2, 2) AS "state_id",
+                    "build_existing_model.create_bar_from_building_type_ratios_bldg_type_a" as "building_type",
+                    "upgrade",
+                    "total_site_gas_kbtu",
+                    "total_site_electricity_kwh"
+                FROM
+                    "{self.timeseries_table}"
+                JOIN "{self.baseline_table}"
+                    ON "{self.timeseries_table}"."building_id" = "{self.baseline_table}"."building_id"
+                WHERE {where_clause}
+            )
+            GROUP BY
+                "upgrade",
+                "month",
+                "state_id",
+                "building_type"
+        """
+
+        return query.strip()
+
+    def get_timeseries_aggregation_query(self,
+                                       upgrade_id: Union[int, str],
+                                       enduses: List[str],
+                                       weight_view_table: str,
+                                       group_by: Optional[List[str]] = None,
+                                       restrictions: List[tuple] = None,
+                                       timestamp_grouping: str = 'hour',
+                                       building_ids: Optional[List[int]] = None,
+                                       include_sample_stats: bool = True,
+                                       include_area_normalized_cols: bool = False) -> str:
+        """
+        Build query for timeseries data aggregation similar to buildstock query functionality.
+
+        This matches the pattern from working buildstock queries that join timeseries data
+        with a weight view table for proper weighting and aggregation.
+
+        Args:
+            upgrade_id: Upgrade ID to filter on
+            enduses: List of end use columns to select
+            weight_view_table: Name of the weight view table (required - e.g., 'rtuadv_v11_md_agg_national_by_state_vu' or 'rtuadv_v11_md_agg_national_by_county_vu')
+            group_by: Additional columns to group by (e.g., ['build_existing_model.building_type'])
+            restrictions: List of (column, values) tuples for filtering
+            timestamp_grouping: How to group timestamps ('hour', 'day', 'month')
+            building_ids: Specific building IDs to filter on (optional)
+            include_sample_stats: Whether to include sample_count, units_count, rows_per_sample
+            include_area_normalized_cols: Whether to include weighted area and kwh_weighted columns for AMI comparison
+
+        Returns:
+            SQL query string
+        """
```

Comment:
really nice docstrings - thank you! I always put an extra newline between 127 and 128 but that's a style thing from when I got to use pycharm like 5 years ago.